src/Storages/StorageFilesystem.cpp

```diff
@@ -0,0 +1,356 @@
+#include <chrono>
+#include <filesystem>
+#include <base/types.h>
+#include <Common/ConcurrentBoundedQueue.h>
+#include <Common/logger_useful.h>
+#include <Interpreters/Context.h>
+#include <Processors/ISource.h>
+#include <Storages/StorageFilesystem.h>
+#include <Storages/StorageFactory.h>
+#include <Storages/checkAndGetLiteralArgument.h>
+
+namespace fs = std::filesystem;
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int DATABASE_ACCESS_DENIED;
+    extern const int DIRECTORY_DOESNT_EXIST;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+String fileTypeToString(fs::file_type type)
+{
+    switch (type)
+    {
+        case fs::file_type::none:
+            return "none";
+        case fs::file_type::not_found:
+            return "not found";
+        case fs::file_type::regular:
+            return "regular file";
+        case fs::file_type::directory:
+            return "directory";
+        case fs::file_type::symlink:
+            return "symlink";
+        case fs::file_type::block:
+            return "block";
+        case fs::file_type::character:
+            return "character";
+        case fs::file_type::fifo:
+            return "fifo";
+        case fs::file_type::socket:
+            return "socket";
+        case fs::file_type::unknown:
+            return "unknown";
+    }
+}
+
+String permissionsToString(fs::perms perms)
```

Comment:
We shouldn't use "mini-languages" instead of structured data.

It should be represented by individual columns, e.g.

Bool user_can_read, user_can_write, user_can_execute, group_can_read, ...

Reply:
I didn't change this part yet from the initial PR. Makes sense